### PR TITLE
Fix compilation with mingw

### PR DESCRIPTION
**Pull request by hardening**
This patch addresses some issues when compiling against mingw. With these changes FreeRDP can be compiled out-of-the box on a Ubuntu 20.04.

### CODE REVIEW

File: winpr/include/winpr/smartcard.h [hunk 1]
```diff
@@ -115,6 +115,14 @@
 
 #endif
 
+/* missing definition with mingw */
+#ifndef SCARD_E_PIN_CACHE_EXPIRED
+#define SCARD_E_PIN_CACHE_EXPIRED (HRESULT)(0x80100032L)
```
Comment (akallabeth) [inline winpr/include/winpr/smartcard.h:120]: shouldn´t each of these be guarded by a `#ifndef`?

**Review by akallabeth — APPROVED** [review]
LGTM

File: winpr/include/winpr/smartcard.h (hunk 1 above)
Comment (hardening) [reply winpr/include/winpr/smartcard.h:120]: Well I think it comes as a full set, but I can split it for more safety.

File: winpr/include/winpr/spec.h [hunk 2]
```diff
@@ -965,6 +965,7 @@ extern "C++"
 
 #endif
 
+#if !defined(__MINGW32__)
```
Comment (akallabeth) [inline winpr/include/winpr/spec.h:968]: Does mingw already define these macros?
Would be no good idea to export all functions for mingw builds

Comment (hardening) [reply winpr/include/winpr/spec.h:968]: yes they are defined in mingw headers.

**Pull request merged**